crates/core/src/db/message_log.rs

```diff
@@ -217,6 +267,124 @@ impl MessageLog {
     }
 }
 
+/// A read-only view of an on-disk [`Segment`] of the [`MessageLog`].
+///
+/// The underlying file is opened lazily when calling [`SegmentView::try_into_iter`]
+/// or [`SegmentView::try_into_file`].
+#[derive(Clone, Debug)]
+pub struct SegmentView {
```

Comment:
Why is this required instead of `MessageLogIter`? Also presumably we should implement the internals of `MessageLogIter` with `SegementView` because we're duplicating some code that really needs to stay in sync.

Reply:
I was going to ask about `MessageLogIter`, and perhaps submit a separate patch: the issue is, it has very different semantics. For one, it panics on I/O errors, which isn't great when it's not a matter of database consistency. More importantly, though, `segments_for_offset` returns the **first** segment if `offset` does not fall into the **last** segment -- the significance of which I don't understand.

So imho this would require to write a number of tests at least, which I wouldn't want to conflate with this patch (which does not change any database internals).

Reply:
> Why is this required instead of MessageLogIter?

It is also not helpful to borrow `MessageLog` when that value is behind a mutex.